packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -473,6 +485,244 @@ export const RemoteServerFactory = {
         })
     },
 
+    /**
+     * @private
+     */
+    _setupBasePathMiddleware(app) {
+        // Cache the express route regexes to avoid re-calculating them on every request.
+        let expressRouteRegexes
+
+        /**
+         * Remove the base path from a path.
+         *
+         * If path is like '/basepath/something', this returns '/something'
+         * If path is exactly '/basepath', this returns '/'
+         * If path doesn't start with base path or if there is no base path defined,
+         * returns the unmodified path
+         *
+         * @param path {string} the path to remove the base path from
+         * @returns {string} the path with the base path removed
+         */
+        const removeBasePathFromPath = (path) => {
+            const basePath = getEnvBasePath()
+            if (!basePath) {
+                return path
+            }
+
+            if (path.startsWith(basePath + '/')) {
+                return path.substring(basePath.length)
+            } else if (path === basePath) {
+                return '/'
+            }
+            return path
+        }
+
+        /**
+         * Produces a regex that corresponds to the express route pattern provided.
+         */
+        const convertExpressRouteToRegex = (routePattern) => {
```

Comment:
let's move this to a util func and out of this file, I don't think it needs to be in this file.